src/main.cpp

```diff
@@ -2493,6 +2525,12 @@ bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &pos, unsigne
     unsigned int nOldChunks = (pos.nPos + UNDOFILE_CHUNK_SIZE - 1) / UNDOFILE_CHUNK_SIZE;
     unsigned int nNewChunks = (nNewSize + UNDOFILE_CHUNK_SIZE - 1) / UNDOFILE_CHUNK_SIZE;
     if (nNewChunks > nOldChunks) {
+        if (fPruneMode) {
+            // nUndoSize has already been incremented, so passing in pos.nPos
+            // would cause Autoprune to overstate disk usage (see related
+            // comment in FindBlockPos).
+            Autoprune(nNewChunks * UNDOFILE_CHUNK_SIZE - vinfoBlockFile[nFile].nUndoSize);
+        }
```

Comment:
In the meantime, you could try commenting out this call to Autoprune.  It should still keep you very close to the target and ought to fix the problem.